torch/_functorch/aot_autograd.py

```diff
@@ -3914,27 +3996,31 @@ def backward(ctx, *flat_args):
             # and we filter them out here before passing the remaining grad_outputs into the compiled backward.
             num_mutated_inps = CompiledFunction.metadata.num_mutated_inputs
             num_intermediate_bases = CompiledFunction.metadata.num_intermediate_bases
+            num_graph_handled_inputs = CompiledFunction.metadata.num_mutated_graph_handled_indices
             expected_grad_outs = (
-                CompiledFunction.metadata.num_outputs + num_mutated_inps + num_intermediate_bases
+                CompiledFunction.metadata.num_outputs + num_mutated_inps + num_intermediate_bases - num_graph_handled_inputs
```

Comment:
Same thing here, just do `inner_meta.num_mutated_runtime_inps` instead of subtracting stuff